prime-dotnet-webapi/Configuration/Database/Unseeded Tables/EnrolleeConfiguration.cs

```diff
@@ -17,6 +17,12 @@ public void Configure(EntityTypeBuilder<Enrollee> builder)
             builder
                 .HasIndex(e => e.HPDID)
                 .IsUnique();
+            builder
+                .HasOne(e => e.PaperEnrolment)
+                .WithOne(l => l.Enrollee);
+            builder
+                .HasOne(e => e.LinkedEnrolment)
+                .WithOne(l => l.PaperEnrollee);
```

Comment:
I'm concerned by the name of these navigation properties; they are too ambiguous in my opinion. For one thing, they are not the linked `Enrollee` objects, they are the `EnrolleeLinkedEnrolment` objects. This is kinda confusing when you have to do `Enrollee,PaperEnrolment.PaperEnrollee`. It is also a bit ambiguous what `Enrollee.LinkedEnrolment` means on a non-paper `Enrollee`. I would expect that regular Enrollees with a linked paper enrolment would have `Enrollee.PaperEnrolment != null`, but then what is `Enrollee.LinkedEnrolment`? They have a linked enrolment, so it sounds like these are redundant properties.

Reply:
> I'm concerned by the name of these navigation properties; they are too ambiguous in my opinion. For one thing, they are not the linked `Enrollee` objects, they are the `EnrolleeLinkedEnrolment` objects. This is kinda confusing when you have to do `Enrollee,PaperEnrolment.PaperEnrollee`. It is also a bit ambiguous what `Enrollee.LinkedEnrolment` means on a non-paper `Enrollee`. I would expect that regular Enrollees with a linked paper enrolment would have `Enrollee.PaperEnrolment != null`, but then what is `Enrollee.LinkedEnrolment`? They have a linked enrolment, so it sounds like these are redundant properties.

Ideally a single property `LinkedEnrolment`on the `Enrollee` model is the best, which just points to the *linked* enrolllee. For regular Enrollee, `LinkedEnrolment` means a paper enrolment. For paper enrollee, `LinkedEnrolment` means the linked regular enrollee.

However EF does not seem to support using a single property to configure two different relationship indicated by different foreignKey(`FK_EnrolleeLinkedEnrolment_Enrollee_EnrolleeId` and `FK_EnrolleeLinkedEnrolment_Enrollee_PaperEnrolleeId` in table `EnrolleeLinkedEnrolment`). I had to use two separate properties to configure the relationships

Reply:
how about `EnrolleeToPaperLink` and `PaperToEnrolleeLink` for names?

Reply:
> 

Done

Reply:
> 

Done